scenes/novoProjeto.scene.js

```diff
@@ -1,77 +1,123 @@
-const { User } = require('../models/user.model');
-const Scene = require('telegraf/scenes/base')
+/* eslint-disable no-plusplus */
+/* eslint-disable func-names */
+/* eslint-disable no-underscore-dangle */
+const winston = require('winston');
+const Scene = require('telegraf/scenes/base');
 const { Markup } = require('telegraf');
 const axios = require('axios');
-const winston = require('winston');
+const { User } = require('../models/user.model');
+const { projectsService } = require('../services/projects.service');
 
 const novoProjetoScene = new Scene('novoprojeto');
 
-let projeto = {};
+let projects = [];
```

Comment:
Qual a função exata dessa variável `projects` no escopo global ?

Reply:
Nesse modulo eu crio uma especie de paginacao que mostra os projetos do usuario de 2 em 2 no chat do telegram, pra isso eu uso o metodo `getNextTwo` que recebe o index do ultimo projeto exibido em tela e uma array de projetos e retorna os dois proximos projetos, e como eu nao consigo chamar os listeners do telegraf e precisava disponibilizar esse dado entre os listeners acabei fazendo isso. Uma possível solucao que eu vejo seria encapsular
```
projects = await projectsService.getProjectsByUserId(user._id);
  projects = projects.map((project) => (Markup.callbackButton(
    project.name,
    JSON.stringify({ value: project.id, isRepo: true }),
  )));
```
em um metodo e chamá-lo sempre que for preciso, mas na hora só consegui pensar na implementacao atual.